DOC: add compatibility for sphinxcontrib-bibtex>=2

hagenw opened the pull request:
the test in https://github.com/sfstoolbox/sfs-python/pull/168 are failing as `sphinxcontrib-bibtex` introduced a breaing change with version 2.0.0 as you have to add another config setting.

This fixes thie issue.

mgeier commented:
I think now the `references.bib` argument can be removed from `references.rst`:

```rst
.. bibliography:: references.bib
    :style: alpha
    :all:
```

Also, I think `:all:` isn't necessary anymore, right?

I'm not sure though, because I cannot check it locally because I'm getting some errors (which may be related to my broken installation).

hagenw commented:
Thanks, `references.bib` is indeed no longer needed. `:all:` includes all references, also not cited once. I removed that option as well.
I was able to execute the code locally and found also one wrong citation to Ahrens2012 that was not using the bibtex plugin, and corrected that as well.

mgeier reviewed (approved):
Thanks!
